# Java 中用户定义的自定义异常

> 原文:[https://www . tutorialandexample . com/user-defined-custom-exceptions-in-Java](https://www.tutorialandexample.com/user-defined-custom-exceptions-in-java)

在本教程中，我们将通过示例讨论用户定义的自定义异常。

## 介绍

在 Java 中，我们已经主动描述了异常类，例如，ArithmeticException、NullPointerException、ArrayoutOfBound 等等。这些内置的异常现在在预先表征的条件下被触发；例如，将一个数除以零会触发 ArithmeticException。在前面的会话中，我们发现了内置的异常，并使用输出屏幕上的 try-catch 块和 throws 解决了它们。

我们可以在 Java 中创建异常类，并利用 try-catch 块来尝试异常。这些例外情况被称为客户特征或定制豁免。

为了理解这个指令，我们应该有关于 try-catch、throw 关键字和异常的基本信息。

异常是程序执行过程中的事件或问题。它会扰乱程序的整个代码，不能完全执行，并立即终止。

在 Java 编程语言中，我们有一个特定的工具来创建异常类的异常派生类。创建我们的异常被称为自定义异常或用户定义的异常。通常，用户定义的异常用于根据用户要求定制异常。从上面的语句中，我们可以说自定义异常正在创建我们的异常，并通过使用“throw”关键字在控制台或输出屏幕上抛出出现的异常。

Java 编程语言中的自定义异常是开发人员为满足应用程序的特定需求而创建的异常。这就是为什么它在 Java 中也被称为用户定义的异常。

例如:

1.**银行申请:**每位愿意开户的客户必须年满 18 岁。但是如果客户的年龄小于 18 岁，那么针对特定需求的程序会抛出一个定制异常，指示建议。

2.**投票年龄:**如果在印度，公民的年龄小于 18 岁，那么针对该特定要求的程序会抛出“无效年龄”作为自定义例外

让我们以一个用户定义的异常为例，它是通过扩展 Exception 类实现的 DemoException。

在下面的例子中，我们已经创建了我们的异常，它从异常类扩展而来。

## 自定义例外的目的

在 Java 编程中，程序执行过程中会出现大量的异常。有时，我们必须创建异常，我们称之为自定义异常。

使用自定义异常背后有特定的原因。以下是一些原因…

*   捕捉现有的几组异常并给予特殊处理。
*   业务逻辑异常:Java 中有许多与自身需求和领域相关的异常。业务逻辑异常与业务逻辑和工作流相关。这对用户或开发者理解该领域的确切问题是有益的。

为了创建我们的异常，我们需要在 Java 中扩展属于 lang 包的异常类。

**举例:**

在本例中，我们将向构造函数传递一个字符串，这是一个异常的超类，它是使用“getMessage()”函数在创建的对象上获得的。

**程序:**

```
 // Demoexception Class that represents user-defined Exception

class DemoException extends Exception {
    public DemoException(String s)
    {
        // it will call the constructor of the parent exception class
        super(s);
    }
}

// A Class that uses the above DemoException
public class Tutorialandexample {
    public static void main(String args[])
    {
        try {
            // Throw an object of user-defined Exception
            throw new DemoException(“Tutorialandexample");
        }
        catch (DemoException ex) {
            System.out.println("Caught an Exception");

            // Displaying the message from DemoException object System.out.println(ex.getMessage());
        }
    }
} 
```

**输出**

```
Caught an Exception
Tutorialandexample 
```

从上面的代码中，我们可以清楚 DemoException 的构造函数需要一个字符串作为它的参数。使用 super()关键字将参数传递给基类异常的构造函数。如果我们不想给参数，就不需要超级键。

**程序:**

```
// Demo Class to represents user-defined exception

class DemoException extends Exception {
}

// A Jtp Class that uses the above DemoException
public class Jtp{
    public static void main(String args[])
    {
        try {
            // It will throw an object of custom Exception
            throw new DemoException();
        }
        catch (DemoException ex) {
            System.out.println("Caught an Exception");
            System.out.println(ex.getMessage());
        }
    }
}
```

**输出**

```
Caught an Exception
null 
```